test/library/standard/Set/setOperatorOrderTest.good

```diff
@@ -0,0 +1,9 @@
+{(a = LHS), (a = RHS other), (a = LHS other)}
+{(a = LHS), (a = RHS other), (a = LHS other)}
```

Comment:
I think it is a bit concerning - if the user was expecting defining `==` like that to mean that they wouldn't have "duplicates" in the set and now they have "duplicates", that seems like it would be wrong?  But I also think deciding to add an element to a set should use the hash and checking every element with `==` regardless of if they hash to the same place seems like it will really slow the operation down?  But also, correctness is more important than performance . . . but on the other hand, it's the user's choice to define their `==` operator like this . . .

I'd definitely bring it up to other folks and I'd be curious what other languages do.  At the very least, it seems reasonable to warn that `==` and the hash function have this relationship when it comes to sets

Reply:
The discussion on https://github.com/chapel-lang/chapel/issues/18653 seemed to lean in the direction of only briefly mentioning that the objects need to hash to the same value as a note in this PR and that views the question of objects being able to be `==` equivalent while hashing to different values a part of https://github.com/chapel-lang/chapel/issues/18577.

With that discussion and your comments in mind, I am inclined not to mention that the objects must also hash to the same values, since it has been argued that if someone is overriding the `==` operator, that should fall into the users responsibility. Also, I will change the test to include values that are both `==` equivalent and hash to the same value, to clear up confusion and be more future-proof as a test, since we may change how `==` and `hash` interact in the future.

Does that sound alright to you?

Reply:
I would add an alternate version of the test like you describe - while it may result in less work in the future to replace this test, if something accidentally changes that behavior I think we would like to know about it (and I think it was valuable you happened to write the test this way, it made us think about a case we wouldn't have considered otherwise)

Reply:
Alright, I've added both tests. Would you mind taking one more approving look over the PR before I merge? Thanks!